app/assets/javascripts/components/timeline/timeline.jsx

```diff
@@ -13,7 +13,7 @@ import CourseLink from '../common/course_link.jsx';
 import Affix from '../common/affix.jsx';
 import EditableRedux from '../high_order/editable_redux';
 
-import CourseActions from '../../actions/course_actions.js';
+import { deleteAllWeeks } from '../../actions/course_actions_redux.js';
```

Comment:
Using `deleteAllWeeks` directly from an import does not work, as it's a redux "action creator" rather than an action. It needs to be connected to `dispatch` in order to work, so instead of importing it here, it should be imported in the connected parent component `TimelineOverview`, added to `mapDispatchToProps` and then passed to `Timeline` as a prop. See `deleteWeek` and other action creators in `TimelineHandler` for the pattern to follow.

I missed this the first time I skimmed the PR, and that lack of an integration test gave me false confidence in the passing tests. Sorry about that.

Reply:
I just added a feature spec for this.

Reply:
@ragesoss In all the other files where we have used course_actions_redux, we have directly used the function. For example: `import { fetchCourse } from './course_actions';`. Why in this case is it required to import the file and not the function?